src/main/java/com/findmyrecycling/fmrenterprise/FindMyRecyclingController.java

```diff
@@ -39,16 +38,21 @@ public String addFacilityPage(Model model) {
         facility.setFacilityId(110L);
         facility.setMaterialId(50L);
         model.addAttribute(facility);
-        return "AddFacility";
+        return "addFacility";
     }
 
-    @RequestMapping("/saveFacility")
-    public String saveFacility(Facility facility) {
+    @PostMapping("/saveFacility")
```

Comment:
I recommend changing this to a post-mapping since you are saving an object through a Thymeleaf template. In addition, the index template expects a model passed to it, and there is none when an object is saved on the add Facility page. Also, it was causing runtime issues when accessing this method from Thymeleaf

Reply:
Agreed.